app/controllers/campaigns_controller.rb

```diff
@@ -7,12 +7,13 @@ def display_skip_to_main_navigation?
     false
   end
 
+  ::CampaignPage::Campaign::WHITELIST.each do |campaign_page|
+    method_name = campaign_page.underscore
+    class_eval "def #{method_name}; end"
```

Comment:
I read the PR code but I didn't identify the reason of creating a method based on the campaign name and use class eval. Can you explain to me th reason of it, please?